### ISSUE TITLE
Mercator Projection does not work on GeoMakie

### ISSUE THREAD

**Issue by wxguy** [MakieOrg/GeoMakie.jl#143]
I am new to Julia but long time user of Python to produce various Geo plots. I recently got intrest in Julia to implement what I did in Python. However, after reading lot of documentation it appears that GeoMakie is somewhat similar or trying to be similar to what `cartopy` does for Python. I am trying to plot a simple NetCDF variable . So I went ahead and tried to plot few of the examples given in the document site. When I changed the projection, nothing seems to work. Here is my Julia info:

```
julia> versioninfo()
Julia Version 1.8.4
Commit 00177ebc4fc (2022-12-23 21:32 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 8 × 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, tigerlake)
  Threads: 1 on 8 virtual cores
Environment:
  LD_GOLD = /home/wxguy/miniconda/bin/x86_64-conda-linux-gnu-ld.gold
```

Here is the code I am trying to execute:

```
using NCDatasets, GeoMakie, CairoMakie, GeoJSON, Shapefile

nc_file = joinpath("/mnt/Data/plot_examples", "air.nc")

println("Reading NetCDF file...")
ncds = NCDatasets.NCDataset(nc_file, "r")

println("Extracting Longitudes...")
lons = ncds["lon"][:]

println("Extracting Latitudes...")
lats = (ncds["lat"][:])

println("Extracting Ait Temp...")
air_temp = ncds["air"][:,:,end]

fig = Figure()
ax = GeoAxis(fig[1,1], 
     coastlines = true, 
     title = "Example NetCDF Plot"
)

cf = GeoMakie.surface!(ax, lons, lats, air_temp; shading = false)
cb = GeoMakie.Colorbar(fig[1, 2], cf)

println("Closing DataSet")
close(ncds)

display(fig)
```

The above code produces this output.
![plot_example](https://user-images.githubusercontent.com/9054645/211055168-cbe1cb4f-cb85-47c4-9bc7-5887ea8c3d30.png)

However, if I change the projection to Metcator with `ax = GeoAxis(fig[1,1], dest = "+proj=merc")` , I get wiered plot as shown below.

![plot_example_error](https://user-images.githubusercontent.com/9054645/211055420-a1fd34a4-2dac-463b-95c5-1055778456c4.png)

Can someone guide me correct documentation or clarify if I am doing it right.

**Comment by visr** [comment]
I'm not sure how we could fix this or what cartopy does here, but it appears to be related to the high distortion of Mercator at the high latitudes. If I look for instance at https://en.wikipedia.org/wiki/Mercator_projection, they only plot latitude between -85 and 85. We can do the same to get a less distorted looking Mercator plot:

```julia
fig = Figure()
ga = GeoAxis(
    fig[1, 1];
    dest = "+proj=merc",
    coastlines = true,
    latlims = (-85, 85),
)
fig
```

**Comment by asinghvi17** [comment]
I think this ties in to the issue that we don't have automated heuristics to determine what range of lon/lat is suitable for a given projection, and I have no idea where Cartopy gets that.  If anyone knows where I can get that data from, please let me know!  Otherwise I think @visr's suggestion is the best we can get :)

**Comment by wxguy** [comment]
I agree with what is mentioned here https://en.wikipedia.org/wiki/Mercator_projection and other literature I read online. Most of the literature suggests that the Mercator projection's latitude limits are -85 to + 85. However, when I looked at the source code of `cartopy` https://scitools.org.uk/cartopy/docs/v0.15/_modules/cartopy/crs.html#Mercator, it appears that they use -80 for south and +84 for north latitudes as limits. I am not sure why they have kept it like that and no info/ reference has been given. When I further read some other literature, they have different limits. A prominent one is `ArcGIS` at 
https://desktop.arcgis.com/en/arcmap/latest/map/projections/mercator.htm which has a range from -89 to + 89. 

It is up to GeoMakie developers to decide on the recommended max/ min latitudes for Mercator projection. However, whatever it may be, **it should have the default key/ value pair as being done by the `cartopy`** so that users are able to plot the map on their first try itself.

**Comment by visr** [comment]
Cartopy has seperate classes on top of common PROJ projections to be able to add these limits. If anyone is aware of a good way to automatically get such kind of limits from PROJ that would be the easiest, since we wouldn't have to maintain that code, and have it working for any projection.

In any case an example like this could go in the docs to help users.
